### ISSUE TITLE
Support functional (stateless) components

### ISSUE THREAD

**Issue by zalmoxisus** [gaearon/react-transform-catch-errors#18]
Doesn't catch errors for React stateless components.
[An example](https://github.com/zalmoxisus/react-transform-boilerplate/commit/aa673781306c1abf1b0c19eef0f0d88c3d8973fa) to reproduce.

**Comment by gaearon** [comment]
Is Babel plugin already supporting functional components?

**Comment by zalmoxisus** [comment]
Not sure whether it supports them completely, but `react-transform-hmr` has no problems with them, that's why I addressed it here. Better to open the issue there?

**Comment by gaearon** [comment]
No, react-transform-hmr is completely unaware of them. The issue is with Babel plugin which currently doesn't see them.

**Comment by zalmoxisus** [comment]
I see. Actually, just tried [the example](https://github.com/zalmoxisus/react-transform-boilerplate), and it works well, only after an error occur in the functional component, it will not be able to hot reload anymore and will fail with `Cannot read property 'componentWillUnmount' of null`.
